src/hotspot/share/classfile/dictionary.cpp

```diff
@@ -424,16 +401,15 @@ void Dictionary::validate_protection_domain(unsigned int name_hash,
   // and protection domain are expected to succeed.
   {
     MutexLocker mu(THREAD, SystemDictionary_lock);
-    int d_index = hash_to_index(name_hash);
-    add_protection_domain(d_index, name_hash, klass,
+    add_protection_domain(THREAD, klass,
                           protection_domain);
   }
 }
 
 // During class loading we may have cached a protection domain that has
 // since been unreferenced, so this entry should be cleared.
 void Dictionary::clean_cached_protection_domains(GrowableArray<ProtectionDomainEntry*>* delete_list) {
```

Comment:
Only mention of `GrowableArray` is here and it's a pointer, so do we really need to include the `growableArray.hpp` header? A forward declaration should be sufficient. Not entirely sure, maybe the templates screw it up.